pkg/cloudprovider/cloudapi/azure/azure_security.go

```diff
@@ -316,29 +316,47 @@ func (computeCfg *computeServiceConfig) buildEffectiveNSGSecurityRulesToApply(ap
 	appliedToGroupNepheControllerName := appliedToGroupID.GetCloudName(false)
 	azurePluginLogger().Info("Building security rules", "applied to security group", appliedToGroupNepheControllerName)
 	for _, rule := range currentNsgSecurityRules {
-		if rule.Properties == nil {
```

Comment:
This function can be simplified, please check.

There are only specific cases where we ignore the rule like,
- rule.Properties == nil 
- *rule.Properties.Priority == vnetToVnetDenyRulePriority
- ruleAddrGroupName, appliedToGroupNepheControllerName

In all the other cases we add the rule to currentNsgIngressRules and currentNsgEgressRules.

Reply:
Done